src/components/new-safe/create/steps/StatusStep/StatusStepper.tsx

```diff
@@ -1,40 +1,91 @@
 import { Box, Step, StepConnector, Stepper, Typography } from '@mui/material'
+import { useEffect } from 'react'
 import css from '@/components/new-safe/create/steps/StatusStep/styles.module.css'
 import EthHashInfo from '@/components/common/EthHashInfo'
 import { SafeCreationStatus } from '@/components/new-safe/create/steps/StatusStep/useSafeCreation'
 import StatusStep from '@/components/new-safe/create/steps/StatusStep/StatusStep'
 import { usePendingSafe } from './usePendingSafe'
+import { useHasFeature } from '@/hooks/useChains'
+import { useWeb3ReadOnly } from '@/hooks/wallets/web3'
+import { FEATURES } from '@/utils/chains'
+
+// TODO: Remove when we update SDK
+const useSyncZkEvmSafeAddress = (status: SafeCreationStatus) => {
+  const shouldFetchAddress = useHasFeature(FEATURES.SAFE_CREATION_FETCH_ADDRESS)
+  const [pendingSafe, setPendingSafe] = usePendingSafe()
+  const web3ReadOnly = useWeb3ReadOnly()
+
+  useEffect(() => {
+    if (
+      !shouldFetchAddress ||
+      status !== SafeCreationStatus.SUCCESS ||
+      !pendingSafe ||
+      !pendingSafe.txHash ||
```

Comment:
If we relay this will not work as we will never have a txHash.

It's okay as long as we first update the SDK before we enable relaying on any of the zkevm chains ;)

Reply:
I thought about this but I don't forsee up allowing relaying from the beginning. We should prioritise upgrading the SDK first.